compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -7385,6 +7417,41 @@ private void reorderNamedArgs(BLangInvocation iExpr, BInvokableSymbol invokableS
         iExpr.requiredArgs = args;
     }
 
+    private BLangDynamicParamExpr createDynamicParamExpression(BLangExpression condition,
+                                                               BLangExpression conditionalArg) {
+        BLangDynamicParamExpr dymaicExpression = new BLangDynamicParamExpr();
+        dymaicExpression.condition = condition;
+        dymaicExpression.conditionalArgument = conditionalArg;
+        return dymaicExpression;
+    }
+
+    private BLangTernaryExpr createTernaryExprNode(BType type, BLangExpression expr, BLangExpression thenExpr,
+                                                   BLangExpression elseExpr, Location pos) {
+        BLangTernaryExpr ternaryExpr = (BLangTernaryExpr) TreeBuilder.createTernaryExpressionNode();
+        ternaryExpr.pos = pos;
+        ternaryExpr.elseExpr = elseExpr;
+        ternaryExpr.thenExpr = thenExpr;
+        ternaryExpr.expr = expr;
+        ternaryExpr.type = type;
+        return ternaryExpr;
+    }
+
+    private BLangIndexBasedAccess createMemberAccessExprNode(BType type, BLangExpression expr,
+                                                             BLangExpression indexExpr, Location pos) {
+        BLangIndexBasedAccess memberAccessExpr = (BLangIndexBasedAccess) TreeBuilder.createIndexBasedAccessNode();
+        memberAccessExpr.pos = pos;
+        memberAccessExpr.expr = expr;
+        memberAccessExpr.indexExpr = indexExpr;
+        memberAccessExpr.type = type;
+        return memberAccessExpr;
+    }
+
+    private BLangExpression createIgnoreExprNode(BType type) {
+        BLangExpression ignoreExpr = new BLangIgnoreExpr();
+        ignoreExpr.type = type;
+        return ignoreExpr;
+    }
```

Comment:
- These need to go in `ASTBuilderUtil`
- There's already two methods to create `BLangIndexBasedAccess` in `ASTBuilderUtil` which we can probably reuse here
- We probably don't need to set pos

Reply:
Fixed.